### DISCUSSION TITLE
infrastructure/proxmox/gpu-passthrough/igpu-passthrough-to-vm/

### DISCUSSION THREAD

**Discussion by giscus** [fire1ce/3os.org#70] [General]
# infrastructure/proxmox/gpu-passthrough/igpu-passthrough-to-vm/

Proxmox iGPU passthrough to VM configuration for Hardware Acceleration (HW Aceleration) for example using video encoding/decoding and Transcoding for series like Plex and Emby.

https://3os.org/infrastructure/proxmox/gpu-passthrough/igpu-passthrough-to-vm/

**Comment by alandev84** [comment]
i tried your method on intel nuc 11, does not truly passthrough. it point that my gfx is meeting error 43. do you know how to resolve this? also, do you know how to dump vbios for the intel igpu? thanks in advance!

**Reply by fire1ce** [reply]
Unfortunately, error 43 is the most hard to debug since its a software restriction. The routes you can take are different Kernel  version for example 5.13, different vm configuration as Machine type and a-lot of patience.

About the vbios of the vgpu - no idea.

**Reply by SerialVelocity** [reply]
@alandev84 Did you get anywhere with this?

**Reply by lalli87** [reply]
Worked the first time for me. I have same 8th gen intel i5 with 630 integrated GPU. Successfully passed to Ubuntu 24.04

**Comment by herotm** [comment]
I am not seeing the RenderD128, all other settings are the same. Do you know how to debug

**Reply by herotm** [reply]
I updated the kernel and see the file now, but monitor is black screen, i passedthru a keyboard and ctrl+alt+end works fine, just no screen

**Reply by fire1ce** [reply]
try using the Serial Terminal for the display.

**Reply by herotm** [reply]
THANk YOU!!

**Reply by fire1ce** [reply]
@herotm worked?

**Reply by herotm** [reply]
yep, all Linux OS works, just not Windows, donno why, i looked thru other forum posts, no luck.

**Reply by fire1ce** [reply]
yeh gpu passthrough are pain!

**Reply by herotm** [reply]
I am not getting sound thru the HDMI port from the IGPU Passthrough

lspci only shows 

00:02.0 VGA compatible controller: Intel Corporation Device 9a60 (rev 01)

I dont see Audio options, are you familiar with this

**Reply by fire1ce** [reply]
I'm not. Never used the audio nor tried to make it work.

**Comment by ironicbadger** [comment]
Could you provide some more specifics on the kernel versions (if they are special) for specific versions of Intel CPUs? The more modern CPUs like 12th gen for example, do they require anything other than mainline?

**Reply by fire1ce** [reply]
Nothing special about kernel. But proxmox has a recored to brake stuff from kernel to kernel. Have no clue about 12gen. I don't have all the hardware in the world to test this methods.

**Comment by SamBudeh** [comment]
I am newb in Proxmox and Linux. I'm getting getting the following when I try to start the VM:

>swtpm_setup: Not overwriting existing state file.
kvm: -device vfio-pci,host=0000:00:02.0,id=hostpci0,bus=ich9-pcie-port-1,addr=0x0: vfio 0000:00:02.0: error getting device from group 1: Invalid argument
Verify all devices in group 1 are bound to vfio-<bus> or pci-stub and not already in use
stopping swtpm instance (pid 3747) due to QEMU startup error
TASK ERROR: start failed: QEMU exited with code 1

This is the output of ```update-initramfs -u -k all```:

>update-initramfs: Generating /boot/initrd.img-5.15.30-2-pve
Running hook script 'zz-proxmox-boot'..
Re-executing '/etc/kernel/postinst.d/zz-proxmox-boot' in new private mount namespace..
No /etc/kernel/proxmox-boot-uuids found, skipping ESP sync.

Thank you for the write-up. Any pointers are appreciated.

**Comment by xja** [comment]
I'm unable to boot the TrueNAS VM after adding iGPU, the error message is:
> kvm: -device vfio-pci,host=0000:00:02.0,id=hostpci1,bus=ich9-pcie-port-2,addr=0x0: vfio 0000:00:02.0: Failed to set up TRIGGER eventfd signaling for interrupt INTX-0: VFIO_DEVICE_SET_IRQS failure: Transport endpoint is not connected
> TASK ERROR: start failed: QEMU exited with code 1

Actually, when i execute `lspci -nnv | grep VGA`, there is no Intel related VGA info, 
```ShellSession
root@pve:~# lspci -nnv | grep VGA
08:00.0 VGA compatible controller [0300]: ASPEED Technology, Inc. ASPEED Graphics Family [1a03:2000] (rev 41) (prog-if 00 [VGA controller])
```

`ASPEED Graphics` is onboard integrated graphics
Host:
|	Name	|	Value	|
|	-	|	-	|
|	CPU(s)	|	8 x Intel(R) Xeon(R) E-2144G CPU @ 3.60GHz (1 Socket)	|
|	Motherboard	|	SuperMicro X11SCH-LN4F	|
|	Kernel Version	|	Linux 5.15.39-4-pve #1 SMP PVE 5.15.39-4 (Mon, 08 Aug 2022 15:11:15 +0200)	|
|	PVE Manager Version	|	pve-manager/7.2-7/d0dd0e85	|

VM:
|	Name	|	Value	|
|	-	|	-	|
|	Memory	|	10.00 GiB	|
|	Processors	|	4 (1 sockets. 4 cores) [host.flags=+aes]	|
|	BIOS	|	Default (SeaBIOS)	|
|	Display	|	Default	|
|	Machine	|	q35	|
|	SCSI Controller	|	Virt10 SCSI	|
|	Hard Disk (scsi0)	|	local-zfs:vm-101-disk-0.size=32G,ssd=1	|
|	Hard Disk (scsi1)	|	local-zfs:vm-101-disk-1.size=512G.ssd=1	|
|	Network Device (net0)	|	virtio=B2:D1:BC:5D:OD:3F.bridge=vmbrO.firewall=1	|
|	PCI Device (hostpci0)	|	0:00:17	|
|	PCI Device (hostpci1)	|	0000:00:02,pcie=1	|

Host BIOS:
|	Name	|	Value	|
|	-	|	-	|
|Intel (VMX) Virtualization Technology|Enabled|
|Internal Graphics|Enabled|
|VT-d|Enabled|

Any suggestions would be greatly appreciated!

**Reply by fire1ce** [reply]
I'm not sure if Intel® UHD Graphics P630 is supported. Try looking for the cases for the specific igpu

**Reply by xja** [reply]
Thanks for your quick response, I found a reddit thread [Enable Mediated Intel iGPU (GVT-g) for VM's in Proxmox (with Plex)](https://www.reddit.com/r/homelab/comments/jyudnn/enable_mediated_intel_igpu_gvtg_for_vms_in/) with the same iGPU. I will try to follow his steps to see if it works.

**Reply by xja** [reply]
Now I have jellyfin in truenas on pve working by following this thread. In order for jellyfin to transcode with iGPU, `Privileged Mode` must be enabled.

**Comment by ricain59** [comment]
Hi,

Tips for vm linux gpu passthrough. Vm on proxmox bios UEFI + vga memory 128mb for console on proxmox for boot (if necessary) + machine i440fx + disk EFI (this configuration works for me wiht i5 10600k UHD630)

**Comment by zx900930** [comment]
Thank you, I can confirm its working on Intel 12 Gen CPU (I'm using 12400 with UHD 730).
And can get the SR-IOV working:
https://www.michaelstinkerings.org/gpu-virtualization-with-intel-12th-gen-igpu-uhd-730/

**Reply by tomshomelab** [reply]
could you share what kernal youre using for proxmox and the vm you got this working on?

**Comment by subbeh** [comment]
Thanks for the guide. I followed all the steps but I don't see renderD128 in /dev/dri (running VM with Ubuntu Server 22.04). Is there any additional step I need to run to make this work? Am I missing a video driver? "lspci -k" doesn't show any drivers loaded for the iGPU

**Comment by mslookup** [comment]
To whom it may help: I followed this Guide to passthrough an Intel iGPU (12th gen) from Proxmox to a VM (Unraid), but Proxmox keeps complaining "No IOMMU detected". 
Stumbled across the solution for my problem at https://forum.level1techs.com/t/proxmox-no-iommu-detected-please-activate-it/179947/16 - needed to append "intel_iommu=on" to the /etc/kernel/cmdline file. After a reboot, everything worked out perfectly for me.

**Comment by imthenachoman** [comment]
Is my assumption correct that once I pass iGPU to a Windows 10 VM, then the Windows 10 VM will show on the monitor connected to the computer? The GPU is passed to the VM, so the VM will output to that video port. Right?

**Comment by ganonms** [comment]
Thank you! This worked great on my first try.

**Comment by JScharloo** [comment]
For those not seeing `RenderD128` or the card at all in `/dev/dri/`. If the card is visible in `lspci -k` the passtrough is successful but the drivers are not loaded. This can be the case when you are running a LTS release in conjunction with a 13th Gen Raptor Lake Intel CPU. Upgrade the kernel to 6.12 using `apt install linux-image-generic-hwe-22.04`, reboot and now the driver should be loaded and the iGPU visible and usable in the guest.

**Reply by ChaconC** [reply]
Thank you! I finally got iGPU passthrough and transcoding working on my VM by upgrading the kernel to 6.2

I have a Chuwi Larkbox X N100 mini PC, adding words for fellow googlers

**Reply by andromidasj** [reply]
Holy cow, you helped end my weekend-long search on getting `/dev/dri` showing up in the VM. I previously did all the steps for iGPU pass-through, but it looks like I needed that kernel upgrade command and a reboot, and everything works perfectly now. Thank you!

**Comment by Mpaulo** [comment]
anyone got this working on AMD ? i'm trying to passtroght a 7950x3d

**Comment by rfuger** [comment]
![50C167E4D2ACA69555BE1B2CE15E3C5C](https://github.com/fire1ce/3os.org/assets/68551351/450d4cbf-936e-45b2-9d49-7c5982e7b162)
已经全部按照你的说明进行设置，升技B760主板 13500T es直通HDMI无输出，究竟是怎么回事？swtpm_setup: Not overwriting existing state file.这个是什么错误？看虚拟机概要好像也启动了，就是显示器没输出啊，igd_debug.log没输出，大小为0

**Reply by jefison** [reply]
同样的错误 1365u

**Comment by batfish1000** [comment]
Thank you very much for this guide. It was the only one that worked for me. There is one other step I needed - and others may also need. I am using an ASUS motherboard and in the BIOS the iGPU is disabled by default. That meant everything worked as described above except I did not see renderD128 . I went to the ASUS BIOS under Advanced Mode/System Agent Configuration and set iGPU to "enable". I restarted the Linux virtual machine and renderD128 now appeared. That meant I could add the line --device=/dev/dri:/dev/dri to my Plex Docker container and enable hardware transcoding.

**Reply by fire1ce** [reply]
You are welcomed. Enjoy!

**Comment by smithj33** [comment]
Good guides, which are very helpful, although you just threw in everything from every internet guide, but most of it is not needed anymore. Just pointing out some things you might want to update.

vfio_virqfd doesn't need to be loaded in modules anymore. It's built in to the kernel and always loaded at boot. 
None of this is needed, video=simplefb:off video=vesafb:off video=efifb:off video=vesa:off disable_vga=1. It was replaced with initcall_blacklist=sysfb_ini, which actually isn't needed either. 
pcie_acs_override=downstream,multifunction Never applies to an Intel iGPU as it will always be in it's own group.
No need to blacklist the i915 GPU either. 
Your Linux iGPU guide shows a screenshot for i440fx (no PCIe), but the screenshot below it shows you using q35 and pcie=1.

**Reply by user6927** [reply]
I would rather say It is not helpful at all. The author does not fully understand the content copied from somewhere. But someone may spend a lot of time to follow that "guide" with no result.

**Comment by stevefxp1** [comment]
I am able to passthrough the iGPU and configure it in Plex(Linux) but do I need the Quicksync drivers installed also? If yes Quicksync is telling me there is no driver installed bc it does not find the iGPU. It finds the default adapter.

**Comment by viper2035** [comment]
Thank you for the guide. I'm using an HP Elitedesk 800 G5 with an Intel i9500t. I was able to passthrough the igpu to a windows 11 VM and install the driver. But I wasn't able to get the video output via DisplayPort inside the VM working. I just get a black screen after the boot. Do you any ideas what to do?

**Reply by fire1ce** [reply]
you won’t be able to use Hardware outputs of the path through device.

**Reply by viper2035** [reply]
That can't be true. I found multiple threads about it online. See: https://forum.proxmox.com/threads/alderlake-direct-passthrough-to-linux-vm-is-working-with-hdmi-output.138049/#post-615315 or here https://forum.proxmox.com/threads/intel-uh630-igp-passthrough-and-hdmi-output-to-host-screen.106909/post-652279

**Reply by fire1ce** [reply]
never managred to make it work. gl

**Comment by ningatz** [comment]
Followed the guide till the end and  can see the GPU with  sudo lspci -nnv | grep VGA but if I ran cd /dev/dri && ls -la  shows that No such file or directory

Any one knows howto fix this?

**Reply by stevefxp1** [reply]
> Followed the guide till the end and can see the GPU with sudo lspci -nnv | grep VGA but if I ran cd /dev/dri && ls -la shows that No such file or directory
> 
> Any one knows howto fix this?

So a few questions...

1) What does your IOMMU group listing look like?
2) Is this a standard PC build with PCIe expansion slots or a mini PC build with an iGPU?
3) If a PC build is your GPU in the first slot?
4) What GPU are you trying to passthrough?

**Reply by ningatz** [reply]
Hi there

1. IOMMU group listing
01:00.0 VGA compatible controller [0300]: Intel Corporation Alder Lake-N [Intel Graphics] [8086:46d4]

2. It is a Mini PC with Intel N150 

3. -

4. It is Intel iGPU

**Reply by stevefxp1** [reply]
> Hi there
> 
> 1. IOMMU group listing
>    01:00.0 VGA compatible controller [0300]: Intel Corporation Alder Lake-N [Intel Graphics] [8086:46d4]
> 2. It is a Mini PC with Intel N150
> 3. 
> 4. It is Intel iGPU

Please provide the full listing of the IOMMU groups. Second please provide a listing of your SystemD or Grub line where you specify IOMMU enablement.

**Reply by ningatz** [reply]
>Please provide the full listing of the IOMMU groups.

```
IOMMU Group 0:
        00:02.0 VGA compatible controller [0300]: Intel Corporation Alder Lake-N [Intel Graphics] [8086:46d4]
IOMMU Group 1:
        00:00.0 Host bridge [0600]: Intel Corporation Alder Lake-N Processor Host Bridge/DRAM Registers [8086:461c]
IOMMU Group 2:
        00:0a.0 Signal processing controller [1180]: Intel Corporation Platform Monitoring Technology [8086:467d] (rev 01)
IOMMU Group 3:
        00:14.0 USB controller [0c03]: Intel Corporation Alder Lake-N PCH USB 3.2 xHCI Host Controller [8086:54ed]
        00:14.2 RAM memory [0500]: Intel Corporation Alder Lake-N PCH Shared SRAM [8086:54ef]
IOMMU Group 4:
        00:15.0 Serial bus controller [0c80]: Intel Corporation Device [8086:54e8]
IOMMU Group 5:
        00:16.0 Communication controller [0780]: Intel Corporation Alder Lake-N PCH HECI Controller [8086:54e0]
IOMMU Group 6:
        00:17.0 SATA controller [0106]: Intel Corporation Alder Lake-N SATA AHCI Controller [8086:54d3]
IOMMU Group 7:
        00:1a.0 SD Host controller [0805]: Intel Corporation Device [8086:54c4]
IOMMU Group 8:
        00:1c.0 PCI bridge [0604]: Intel Corporation Alder Lake-N PCI Express Root Port #7 [8086:54be]
IOMMU Group 9:
        00:1d.0 PCI bridge [0604]: Intel Corporation Alder Lake-N PCI Express Root Port #9 [8086:54b0]
IOMMU Group 10:
        00:1d.2 PCI bridge [0604]: Intel Corporation Alder Lake-N PCI Express Root Port #11 [8086:54b2]
IOMMU Group 11:
        00:1f.0 ISA bridge [0601]: Intel Corporation Alder Lake-N PCH eSPI Controller [8086:5481]
        00:1f.3 Audio device [0403]: Intel Corporation Alder Lake-N PCH High Definition Audio Controller [8086:54c8]
        00:1f.4 SMBus [0c05]: Intel Corporation Alder Lake-N SMBus [8086:54a3]
        00:1f.5 Serial bus controller [0c80]: Intel Corporation Alder Lake-N SPI (flash) Controller [8086:54a4]
IOMMU Group 12:
        01:00.0 Network controller [0280]: Realtek Semiconductor Co., Ltd. RTL8852BE PCIe 802.11ax Wireless Network Controller [10ec:b852]
IOMMU Group 13:
        02:00.0 Non-Volatile memory controller [0108]: Realtek Semiconductor Co., Ltd. RTS5765DL NVMe SSD Controller (DRAM-less) [10ec:5765] (rev 01)
IOMMU Group 14:
        03:00.0 Ethernet controller [0200]: Intel Corporation Ethernet Controller I226-V [8086:125c] (rev 04)
```

>Second please provide a listing of your SystemD or Grub line where you specify IOMMU enablement.

```
GRUB_CMDLINE_LINUX_DEFAULT="quiet intel_iommu=on iommu=pt pcie_acs_override=downstream,multifunction nofb nomodeset video=vesafb:off,efifb:off"
```

**Reply by stevefxp1** [reply]
> > Please provide the full listing of the IOMMU groups.
> 
> ```
> IOMMU Group 0:
>         00:02.0 VGA compatible controller [0300]: Intel Corporation Alder Lake-N [Intel Graphics] [8086:46d4]
> IOMMU Group 1:
>         00:00.0 Host bridge [0600]: Intel Corporation Alder Lake-N Processor Host Bridge/DRAM Registers [8086:461c]
> IOMMU Group 2:
>         00:0a.0 Signal processing controller [1180]: Intel Corporation Platform Monitoring Technology [8086:467d] (rev 01)
> IOMMU Group 3:
>         00:14.0 USB controller [0c03]: Intel Corporation Alder Lake-N PCH USB 3.2 xHCI Host Controller [8086:54ed]
>         00:14.2 RAM memory [0500]: Intel Corporation Alder Lake-N PCH Shared SRAM [8086:54ef]
> IOMMU Group 4:
>         00:15.0 Serial bus controller [0c80]: Intel Corporation Device [8086:54e8]
> IOMMU Group 5:
>         00:16.0 Communication controller [0780]: Intel Corporation Alder Lake-N PCH HECI Controller [8086:54e0]
> IOMMU Group 6:
>         00:17.0 SATA controller [0106]: Intel Corporation Alder Lake-N SATA AHCI Controller [8086:54d3]
> IOMMU Group 7:
>         00:1a.0 SD Host controller [0805]: Intel Corporation Device [8086:54c4]
> IOMMU Group 8:
>         00:1c.0 PCI bridge [0604]: Intel Corporation Alder Lake-N PCI Express Root Port #7 [8086:54be]
> IOMMU Group 9:
>         00:1d.0 PCI bridge [0604]: Intel Corporation Alder Lake-N PCI Express Root Port #9 [8086:54b0]
> IOMMU Group 10:
>         00:1d.2 PCI bridge [0604]: Intel Corporation Alder Lake-N PCI Express Root Port #11 [8086:54b2]
> IOMMU Group 11:
>         00:1f.0 ISA bridge [0601]: Intel Corporation Alder Lake-N PCH eSPI Controller [8086:5481]
>         00:1f.3 Audio device [0403]: Intel Corporation Alder Lake-N PCH High Definition Audio Controller [8086:54c8]
>         00:1f.4 SMBus [0c05]: Intel Corporation Alder Lake-N SMBus [8086:54a3]
>         00:1f.5 Serial bus controller [0c80]: Intel Corporation Alder Lake-N SPI (flash) Controller [8086:54a4]
> IOMMU Group 12:
>         01:00.0 Network controller [0280]: Realtek Semiconductor Co., Ltd. RTL8852BE PCIe 802.11ax Wireless Network Controller [10ec:b852]
> IOMMU Group 13:
>         02:00.0 Non-Volatile memory controller [0108]: Realtek Semiconductor Co., Ltd. RTS5765DL NVMe SSD Controller (DRAM-less) [10ec:5765] (rev 01)
> IOMMU Group 14:
>         03:00.0 Ethernet controller [0200]: Intel Corporation Ethernet Controller I226-V [8086:125c] (rev 04)
> ```
> 
> > Second please provide a listing of your SystemD or Grub line where you specify IOMMU enablement.
> 
> ```
> GRUB_CMDLINE_LINUX_DEFAULT="quiet intel_iommu=on iommu=pt pcie_acs_override=downstream,multifunction nofb nomodeset video=vesafb:off,efifb:off"
> ```

And your cmdline contains the 3 modules for pass-through? Lastly are you looking to fully pass-through the iGPU or do you want to mediate it?

**Reply by ningatz** [reply]
Yes, aready added the module loaded and updated initramfs
![SCR-20250331-ukvk](https://github.com/user-attachments/assets/8c4ec34b-04fa-4b0e-b735-6e7429468e02)

>Lastly are you looking to fully pass-through the iGPU or do you want to mediate it?

I want to fully pass-through the iGPU

**Reply by stevefxp1** [reply]
> Yes, aready added the module loaded and updated initramfs ![SCR-20250331-ukvk](https://private-user-images.githubusercontent.com/9915381/428776902-8c4ec34b-04fa-4b0e-b735-6e7429468e02.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM0NjQ4MDEsIm5iZiI6MTc0MzQ2NDUwMSwicGF0aCI6Ii85OTE1MzgxLzQyODc3NjkwMi04YzRlYzM0Yi0wNGZhLTRiMGUtYjczNS02ZTc0Mjk0NjhlMDIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMzMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMzFUMjM0MTQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWRlY2FlMWViY2RlZmMyZWY3NWYwMzJhYzRhMmZjZWFkZDQxYmM4Yzk1ZDdjMWRkMWM1ZWYzOGUxYWQ3ZjYzMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.wa0wQzRrf1N_XTiyQ1xwHdIwFqEiVkcpqM1rZUt4W_M)
> 
> > Lastly are you looking to fully pass-through the iGPU or do you want to mediate it?
> 
> I want to fully pass-through the iGPU

I use the following:

> Yes, aready added the module loaded and updated initramfs ![SCR-20250331-ukvk](https://private-user-images.githubusercontent.com/9915381/428776902-8c4ec34b-04fa-4b0e-b735-6e7429468e02.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM0NjQ4MDEsIm5iZiI6MTc0MzQ2NDUwMSwicGF0aCI6Ii85OTE1MzgxLzQyODc3NjkwMi04YzRlYzM0Yi0wNGZhLTRiMGUtYjczNS02ZTc0Mjk0NjhlMDIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMzMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMzFUMjM0MTQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWRlY2FlMWViY2RlZmMyZWY3NWYwMzJhYzRhMmZjZWFkZDQxYmM4Yzk1ZDdjMWRkMWM1ZWYzOGUxYWQ3ZjYzMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.wa0wQzRrf1N_XTiyQ1xwHdIwFqEiVkcpqM1rZUt4W_M)
> 
> > Lastly are you looking to fully pass-through the iGPU or do you want to mediate it?
> 
> I want to fully pass-through the iGPU

What does your vm hardwarwe config look like?

**Reply by ningatz** [reply]
>I use the following:

Am I missing something? I do no see 

>What does your vm hardwarwe config look like?

Here
![SCR-20250401-hjqs](https://github.com/user-attachments/assets/72d710ef-68b6-4b84-bcb5-c0024557009e)

**Comment by stevefxp1** [comment]
Yes everything you have shown me looks to be in good shape. Does your vm not boot? What error are you seeing when it boots?

I have an 8th gen Intel NUC that I am passing through the iGPU, but I am doing it as a mediated passthrough. My configs are attached. I also have two Lenovo M720q PCs that are running my firewall VMs, and my 4 port ethernet adapter is fully passed through. These configs are also attached.

[cmdline.txt](https://github.com/user-attachments/files/19552777/cmdline.txt)
[modules.txt](https://github.com/user-attachments/files/19552778/modules.txt)
[cmdline.txt](https://github.com/user-attachments/files/19552815/cmdline.txt)
[modules.txt](https://github.com/user-attachments/files/19552816/modules.txt)

**Comment by MrLumarico** [comment]
I'm getting an issue where the VM just hangs when it gets to the os boot logo, but no errors, am i doing something wrong?

**Reply by ningatz** [reply]
What do you mean by the VM hangs? Are you referring to the Console not working?

**Reply by MrLumarico** [reply]
i eventually gave up, but what happened was the vm would get to the os display level in the console, and just lock up, become unresponsive and not go any further, output would just freeze

**Comment by stevefxp1** [comment]
I had this bc I tried using Q35 and it didnt like it. Not sure why you would be having this issue.

**Comment by christthegreat** [comment]
Thanks for the tutorial, it is well explained. However, mine's not working :(. I have a Dell Optiplex 7060, with an i5-8500T. VT-D for direct IO is enable, my VM can see the intel 630, but error code 43. The difference is that I am running Proxmox 9.0.4 (kernel 6.14).

I tried changing the Machine to qQ35-6.1 or 6.2, tried 8 9, instead of the default 10.0, still not working :(

**Comment by inventory69** [comment]
After upgrading to Proxmox 9, my vm which had this running sucessfully, doesnt work anymore and when i want to create a new ubuntu vm with the pci device added, the vm stucks at the boot logo.
tried it with arch vm and it seemed to work properly but hw acceleration was missing completely and i didnt get it to work.
using also Dell Optiplex 7060, with an i5-8500T and intel 630 uhd graphics.
Do i need to change something for proxmox 9 or what could be the culprit here?

**Comment by admarty** [comment]
@inventory69  Almost all guide about Proxmox full iGPU passthrough on the internet are outdated.

Follow this guide instead https://github.com/LongQT-sea/intel-igpu-passthru

I have a fresh Proxmox 9 install. It just works without needing to edit the kernel command line or add anything to /etc/modprobe or /etc/modules.

**Reply by inventory69** [reply]
Thanks, that did the job :)
